discovery-provider/src/tasks/index_spl_token.py

```diff
@@ -147,20 +154,42 @@ def parse_spl_token_transaction(
         if error:
             return None
 
+        has_transfer_checked_instruction = has_instruction(
+            meta, TRANSFER_CHECKED_INSTRUCTION
+        )
+        if not has_transfer_checked_instruction:
+            logger.info(
+                f"index_spl_token.py | {tx_sig} No transfer checked instruction found"
+            )
+            return None
+        tx_message = result["transaction"]["message"]
+        instruction = get_valid_instruction(tx_message, meta, SPL_TOKEN_ID)
+        if not instruction:
+            logger.error(
+                f"index_spl_token.py | {tx_sig} No valid instruction for spl token program found"
+            )
+            return None
+
         memo_encoded = parse_memo_instruction(result)
         vendor = decode_memo_and_extract_vendor(memo_encoded) if memo_encoded else None
 
-        sender_root_account = get_solana_tx_owner(meta, SENDER_ACCOUNT_INDEX)
-        receiver_root_account = get_solana_tx_owner(meta, RECEIVER_ACCOUNT_INDEX)
-        account_keys = result["transaction"]["message"]["accountKeys"]
-        receiver_token_account = account_keys[RECEIVER_ACCOUNT_INDEX]
-        sender_token_account = account_keys[SENDER_ACCOUNT_INDEX]
-        prebalance, postbalance = get_solana_tx_token_balances(
-            meta, RECEIVER_ACCOUNT_INDEX
-        )
-        # Skip if there is no balance change.
+        sender_idx = get_account_index(instruction, SENDER_ACCOUNT_INDEX)
+        receiver_idx = get_account_index(instruction, RECEIVER_ACCOUNT_INDEX)
+        account_keys = tx_message["accountKeys"]
+        sender_token_account = account_keys[sender_idx]
```

Comment:
(Don't necessarily have to do this now) I think the helper that would really be valuable is one where you pass in the account index and it pulls out the proper account for you (does both of these operations in a single fn)

Reply:
Great idea but will punt for now.